generation/maisi/scripts/sample.py

```diff
@@ -1006,15 +1026,17 @@ def find_closest_masks(self, num_img):
             raise ValueError("Cannot find body region with given organ list.")
         return final_candidates
 
-    def quality_check(self, image):
+    def quality_check(self, image_data, label_data):
         """
-        Perform a quality check on the generated image. This version disabled quality check and always return True.
-
+        Perform a quality check on the generated image.
         Args:
-            image (torch.Tensor): The generated image.
-
+            image_data (np.ndarray): The generated image.
+            label_data (np.ndarray): The corresponding whole body mask.
         Returns:
             bool: True if the image passes the quality check, False otherwise.
         """
-        # This version disabled quality check
+        outlier_results = is_outlier(self.median_statistics, image_data, label_data, self.label_int_dict)
+        for label, result in outlier_results.items():
+            if result.get("is_outlier", False):
+                return False
```

Comment:
The `quality_check` method can be improved by providing more informative error messages when the quality check fails. Also we can consider logging the specific label and median value that caused the failure.

```python
def quality_check(self, image_data, label_data):
    ...
    for label, result in outlier_results.items():
        if result.get("is_outlier", False):
            logging.debug(f"Quality check failed for label {label} with median value {result['median_value']}")  # or logging.info
            return False
    ...
```

Reply:
Good point. Just added it. Thanks

Reply:
Thanks!